cpp/IRCCmds.cpp

```diff
@@ -226,3 +226,67 @@ void	KickCommand::execute(const std::string &params, IRCClient &client)
 
 	}
 }
+
+void	InviteCommand::execute(const std::string &params, IRCClient &client)
+{
+	std::istringstream	sstring(params);
+	std::string			nick, chanName;
+
+	sstring >> nick >> chanName;
+	if (sstring.fail())
+	{
+		client.sendMessage(ERR_NEEDMOREPARAMS(client.getNick(), "INVITE"));
+		return ;
+	}
+	if (!client.nickAlreadyInUse(nick, client.getFd()))
+	{
+		client.sendMessage(ERR_NOSUCHNICK(client.getNick(), nick));
+		return ;
+	}
+	if (!client.channelNameInUse(chanName))
+	{
+		client.sendMessage(ERR_NOSUCHCHANNEL(client.getNick(), chanName));
+		return ;
+	}
+	if (!client.getOp(chanName))
+	{
+		client.sendMessage(ERR_CHANOPRIVSNEEDED(client.getNick(), chanName));
+		return ;
+	}
+	client.getClient(nick).sendMessage(":" + client.getNick() + " INVITE " + nick + " " + chanName + "\r\n");
+}
+
+void TopicCommand::execute(const std::string &params, IRCClient &client)
+{
+    std::istringstream sstring(params);
+    std::string chanName, topic;
+
+    sstring >> chanName;
+    if (sstring.fail())
+    {
+        client.sendMessage(ERR_NEEDMOREPARAMS(client.getNick(), "TOPIC"));
+        return;
+    }
+    const IRCServer* server = client.getServer();
```

Comment:
ici je comprend la demarche mais ce que tu dois faire c'est compartimenter le code :
      const IRCServer* server = client.getServer();
      std::map<std::string, IRCChannel>::const_iterator it = server->getChannels().find(chanName);
tu dois cree cette fonction dans la classe client de une ca permetterea de la reutilser et de deux c'est ce le code restera homogene car c'est de cette manierre que on la fais avec sefir de plus tu pourra suprimer le getter sur serveur car tu y aura acces dans la class client